cirq/ops/pauli_string.py

```diff
@@ -44,7 +44,12 @@ def __init__(self,
                  coefficient: Union[int, float, complex] = 1) -> None:
         if qubit_pauli_map is None:
             qubit_pauli_map = {}
-        for p in qubit_pauli_map.values():
+        qubit_pauli_map = {
+            q: p
+            for q, p in qubit_pauli_map.items()
+            if not isinstance(p, common_gates.IdentityGate)
+        }
+        for q, p in qubit_pauli_map.items():
```

Comment:
If you restore `for p in qubit_pauli_map.values();` then linter will be happy :-)

Reply:
whoops! I changed this in an attempt to do the filtering within the loop but python (and everyone) is very unhappy if you remove things from a dictionary while iterating over it

Reply:
Yeah, too many moving parts confuse the python interpreter.